include/mbedtls/config_adjust_legacy_crypto.h

```diff
@@ -360,45 +360,9 @@
 #endif /* MBEDTLS_PSA_CRYPTO_C */
 
 /* Helpers to state that each key is supported either on the builtin or PSA side. */
-#if defined(MBEDTLS_ECP_DP_SECP521R1_ENABLED) || defined(PSA_WANT_ECC_SECP_R1_521)
-#define MBEDTLS_ECP_HAVE_SECP521R1
-#endif
-#if defined(MBEDTLS_ECP_DP_BP512R1_ENABLED) || defined(PSA_WANT_ECC_BRAINPOOL_P_R1_512)
-#define MBEDTLS_ECP_HAVE_BP512R1
-#endif
-#if defined(MBEDTLS_ECP_DP_CURVE448_ENABLED) || defined(PSA_WANT_ECC_MONTGOMERY_448)
-#define MBEDTLS_ECP_HAVE_CURVE448
-#endif
-#if defined(MBEDTLS_ECP_DP_BP384R1_ENABLED) || defined(PSA_WANT_ECC_BRAINPOOL_P_R1_384)
-#define MBEDTLS_ECP_HAVE_BP384R1
-#endif
-#if defined(MBEDTLS_ECP_DP_SECP384R1_ENABLED) || defined(PSA_WANT_ECC_SECP_R1_384)
-#define MBEDTLS_ECP_HAVE_SECP384R1
-#endif
-#if defined(MBEDTLS_ECP_DP_BP256R1_ENABLED) || defined(PSA_WANT_ECC_BRAINPOOL_P_R1_256)
-#define MBEDTLS_ECP_HAVE_BP256R1
-#endif
-#if defined(MBEDTLS_ECP_DP_SECP256K1_ENABLED) || defined(PSA_WANT_ECC_SECP_K1_256)
-#define MBEDTLS_ECP_HAVE_SECP256K1
-#endif
-#if defined(MBEDTLS_ECP_DP_SECP256R1_ENABLED) || defined(PSA_WANT_ECC_SECP_R1_256)
-#define MBEDTLS_ECP_HAVE_SECP256R1
-#endif
-#if defined(MBEDTLS_ECP_DP_CURVE25519_ENABLED) || defined(PSA_WANT_ECC_MONTGOMERY_255)
-#define MBEDTLS_ECP_HAVE_CURVE25519
-#endif
 #if defined(MBEDTLS_ECP_DP_SECP224K1_ENABLED) || defined(PSA_WANT_ECC_SECP_K1_224)
```

Comment:
Not sure why we keep this one, pending PR?

Reply:
From discussion with @gabor-mezei-arm, it cannot be enabled in PSA. Correct me if I'm wrong!

Reply:
See #3541

Reply:
secp224k1 will be removed in 4.0, both because there won't be a legacy API to access it and because we're removing all curves ≤225 bits. If it helps to remove it now, feel free.

Reply:
Thanks @gilles-peskine-arm. I don't think we are going to remove it as part of this work but just don't make a special case. Thus @eleuzi01 as for the other curves please replace MBEDTLS_ECP_HAVE_SECP224K1 with PSA_WANT_ECC_SECP_K1_224 in the code base and eventually remove MBEDTLS_ECP_HAVE_SECP224K1.

Reply:
You might run into trouble with test scripts though. If you do, please don't add workarounds, just remove secp224k1.

Reply:
thanks for the heads-up, let's see how it goes.

Reply:
I agree with the strategy of initially treating secp224k1 the same as other curves, but if if start giving us even the slightest trouble, let's just remove it already. (Indeed doing all of #8136 may be more work, but I expect removing just secp224k1 should be quite easy.)

Reply:
Done in #9544, CI is not complaining